src/app/shared/episode-select/episode-select-dropdown.component.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, ElementRef, Input, Renderer2, OnInit, ViewChild } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Episode, EPISODE_SELECT_PAGE_SIZE } from '../../ngrx';
+import * as ACTIONS from '../../ngrx/actions';
+
+@Component({
+  selector: 'metrics-episode-select-dropdown',
+  template: `
+    <div class="dropdown" [class.open]="open">
+      <div class="overlay" (click)="toggleOpen()"></div>
+      <div class="dropdown-button" *ngIf="totalEpisodes !== null">
+        <button (click)="toggleOpen()">
+          <span class="button-text">
+            {{ totalEpisodes | i18nPlural: {'=0': 'No episodes', '=1' : '1 episode', 'other' : '# episodes'} }}
+          </span>
+          <span class="down-arrow"></span>
+        </button>
+      </div>
+      <div class="dropdown-content rollout" #dropdownContent>
+        <metrics-episode-search
+          [searchTerm]="searchTerm"
+          [searchTotal]="searchTotal"
+          (search)="loadEpisodesOnSearch($event)">
+        </metrics-episode-search>
+        <metrics-episode-select-list
+          [episodes]="episodes"
+          [episodesLoading]="episodesLoading"
+          [selectedEpisodes]="selectedEpisodes"
+          [totalEpisodes]="totalEpisodes"
+          (selectEpisode)="onToggleSelectEpisode($event)">
+        </metrics-episode-select-list>
+      </div>
+    </div>
+  `,
+  styleUrls: ['../dropdown/dropdown.css', '../dropdown/nav-dropdown.css']
+})
+export class EpisodeSelectDropdownComponent implements OnInit {
+  @Input() episodes: Episode[];
+  @Input() searchTerm: string;
+  @Input() episodesLoading: boolean;
+  @Input() selectedEpisodes: string[];
+  @Input() totalEpisodes: number;
+  @Input() searchTotal: number;
+  @Input() lastPage: number;
+  @Input() maxPages: number;
+  @Input() podcastId: string;
+  @ViewChild('dropdownContent') dropdownContent: ElementRef;
+  open = false;
+
+  constructor(private renderer: Renderer2,
+              private store: Store<any>) {}
+
+  ngOnInit() {
+    this.loadEpisodesOnScroll();
+  }
+
+  loadEpisodesOnScroll() {
+    const element = this.dropdownContent.nativeElement;
+    this.renderer.listen(element, 'scroll', () => {
+      // when scrollTop + clientHeight approaches scrollHeight, load the next page
+      const { clientHeight, scrollHeight, scrollTop } = element;
+      if (!this.episodesLoading &&
+          scrollTop + clientHeight >= (scrollHeight - 100) &&
+          this.lastPage + 1 <= this.maxPages) {
+        this.loadEpisodes(this.lastPage + 1, this.searchTerm);
+        this.store.dispatch(new ACTIONS.GoogleAnalyticsEventAction({gaAction: 'episode-select-page-load', value: this.lastPage + 1}));
+      }
+    });
+  }
+
+  loadEpisodesOnSearch(search: string) {
+    this.loadEpisodes(1, search);
```

Comment:
Does dispatching 2 separate actions make the most sense here? My gut says to roll them up into 1 but when I think about it I can't see any downside to doing it this way.

Reply:
I suppose the google analytics could be handled as a (side) effect when CastleEpisodeSelectPageLoadAction is dispatched, have an effect to dispatch the GoogleAnalyticsEventAction (which is itself an effect actually, `fromGAEvent$`) but it feels more direct to me to just dispatch both actions at their source. Mike Ryan did a great talk at ng-conf last year about designing Actions if you wanna learn more about their thoughts: https://www.youtube.com/watch?v=JmnsEvoy-gY

Reply:
Yeah, thinking about it more I think that's correct. Moving it to the side effect level means that it's a layer of abstraction away from the user interaction, which is really what you're tracking. I'll check out that talk.